result.go

```diff
@@ -304,11 +304,15 @@ func (rs resultSingle) DotResult() []*dot.Result {
 	return dotResults
 }
 
-func (rs resultSingle) Extract(cw containerWriter, v reflect.Value) {
-	cw.setValue(rs.Name, rs.Type, v)
+func (rs resultSingle) Extract(cw containerWriter, decorated bool, v reflect.Value) {
+	if decorated {
+		cw.setDecoratedValue(rs.Name, rs.Type, v)
+	} else {
+		cw.setValue(rs.Name, rs.Type, v)
```

Comment:
We can do something like this to reduce nesting. https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting

```suggestion
func (rs resultSingle) Extract(cw containerWriter, decorated bool, v reflect.Value) {
	if decorated {
		return cw.setDecoratedValue(rs.Name, rs.Type, v)
	}
	cw.setValue(rs.Name, rs.Type, v)
```